Code/GraphMol/Atom.cpp

```diff
@@ -329,6 +329,21 @@ int Atom::calcImplicitValence(bool strict) {
   if (d_explicitValence == -1) {
     this->calcExplicitValence(strict);
   }
+  // special cases
+  if (d_atomicNum == 0) {
+    d_implicitValence = 0;
+    return 0;
+  }
+  if (d_atomicNum == 1 && d_numRadicalElectrons == 0) {
+    if (d_formalCharge == 1 || d_formalCharge == -1) {
+      d_implicitValence = 0;
+      return 0;
+    } else if (d_formalCharge == 0) {
+      d_implicitValence = 1;
+      return 1;
```

Comment:
why not
```
if (d_formalCharge == 0) {
  d_implicitValence = 1;
  return 1;
} else {
  d_implicitValence = 0;
  return 0;
}
```

I'm assuming d_formalCharge > 1 is nonsensical.

Reply:
yeah, it's nonsensical, so it falls through and generates an error later (there's a test for this)
I will make that more explicit here